packages/next/pages/_document.tsx

```
@@ -71,10 +72,73 @@ function getPolyfillScripts(context: HtmlProps, props: OriginProps) {
     ))
 }
 
+function getPreNextWorkerScripts(context: HtmlProps, props: OriginProps) {
+  const { scriptLoader, crossOrigin, optimizeScripts } = context
+
+  if (!optimizeScripts) return null
+
+  try {
+    let {
+      partytownSnippet,
+    } = require(/* webpackIgnore: true */ '@builder.io/partytown/integration'!)
+
+    const children = Array.isArray(props.children)
+      ? props.children
+      : [props.children]
+
+    // Check to see if the user has defined their own Partytown configuration
+    const userDefinedConfig = children.find(
+      (child: React.ReactElement) =>
+        child?.props?.dangerouslySetInnerHTML?.__html.length &&
+        'data-partytown-config' in child.props
+    )
+
+    return (
+      <>
+        {!userDefinedConfig && (
+          <script
+            data-partytown-config=""
+            dangerouslySetInnerHTML={{
+              __html: `
+            partytown = {
+              lib: "/_next/static/~partytown/"
+            };
+          `,
+            }}
+          />
+        )}
+        <script
+          data-partytown=""
+          dangerouslySetInnerHTML={{
+            __html: partytownSnippet(),
+          }}
+        />
+        {(scriptLoader.worker || []).map((file: ScriptProps, index: number) => {
+          const { strategy, ...scriptProps } = file
+          return (
+            <script
+              {...scriptProps}
+              type="text/partytown"
+              key={scriptProps.src || index}
+              nonce={props.nonce}
+              data-nscript="worker"
+              crossOrigin={props.crossOrigin || crossOrigin}
+            />
+          )
+        })}
+      </>
+    )
+  } catch {
```

Do we want to suppress the error here?